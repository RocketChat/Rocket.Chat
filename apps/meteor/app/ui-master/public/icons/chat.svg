<svg id="chat" viewBox="0 0 20 20">
  <path d="M3.36312881,15.7593532 C3.32888874,15.7771726 3.29959994,15.7925621 3.27370934,15.8064174 C3.73031444,16.1917386 4.92393678,16.4549204 7.10274479,16.4476544 L7.21881505,16.4476539 C8.55574002,16.4476543 9.6998097,16.021859 10.6510241,15.170268 L12.188241,15.2712054 C11.5513943,15.8887667 11.2142468,16.2130071 11.1767984,16.2439265 C10.1491488,17.09241 8.76567843,17.5637907 7.21881505,17.5637907 L7.10806092,17.5644215 L6.97721254,17.5644231 C4.30241208,17.5644231 2.62246666,17.1951018 2.10341288,16.0930165 C1.82161029,15.4951149 2.13151102,15.1455492 2.84114675,14.7722012 C3.31318797,14.5227311 3.32172914,14.5103228 3.24657386,14.3473099 C3.20302404,14.2436958 3.16526615,14.1604157 3.07972552,13.9763174 C2.74839464,13.2684751 2.62592155,12.9023078 2.62592155,12.4061039 C2.62592155,11.562445 2.83045155,10.7486059 3.21214892,10.0211892 C3.42204019,9.66099218 3.78721597,9.06536542 4.270822,8.82818513 C4.270822,8.80275057 4.53792399,10.0211892 4.53792399,10 C4.45304843,10.1183326 4.36401443,10.2496015 4.270822,10.3938065 C3.78721597,11.0991197 3.7445882,11.6631394 3.7445882,12.4061039 C3.7445882,12.6985231 3.82924269,12.9394008 4.09330148,13.5047226 C4.19543836,13.7251753 4.22995648,13.801243 4.270822,13.8994264 C4.58782821,14.5810467 4.36137367,15.1036933 3.78721597,15.5080098 C3.65916127,15.5981847 3.55361796,15.6588423 3.36312881,15.7593532 Z M16.5696621,12.3226926 C16.5350205,12.3039671 16.4951623,12.2829692 16.4468239,12.2578135 C16.2265134,12.1415693 16.1048536,12.0716493 15.9574055,11.9678178 C15.2986909,11.5039571 15.0399564,10.9068098 15.4049351,10.1218159 C15.4504792,10.0121781 15.4903754,9.92425877 15.608285,9.66976084 C15.914896,9.01333804 16.0132801,8.73339397 16.0132801,8.39198996 C16.0132801,5.80517045 13.90418,3.70028722 11.3120655,3.70028722 C8.71995111,3.70028722 6.61085101,5.80517045 6.61085101,8.39198996 C6.61085101,11.1096858 8.86296092,13.082961 11.9786225,13.082961 L12.112848,13.0829616 C14.6633683,13.0914671 16.0501708,12.7797435 16.5696621,12.3226926 Z M16.5612822,10.6280948 C16.4685253,10.8294269 16.4833349,10.8509584 17.0358533,11.1429611 C17.8499187,11.5712649 18.2032866,11.969896 17.8824254,12.6506683 C17.2857779,13.9175051 15.3458564,14.3439799 12.2581228,14.3439799 L12.1073975,14.3439799 L11.9780624,14.3432467 C8.14202747,14.3432482 5.34836876,11.8401459 5.34836876,8.39198996 C5.34836876,5.10955211 8.0234498,2.44 11.3120655,2.44 C14.6008168,2.44 17.276494,5.10968724 17.276494,8.39198996 C17.276494,8.96303774 17.1353362,9.38506819 16.7528285,10.2022436 C16.6539056,10.4151439 16.6101724,10.5116034 16.5612822,10.6280948 Z"/>
</svg>