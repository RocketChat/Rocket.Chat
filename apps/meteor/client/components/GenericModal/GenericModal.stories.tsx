import { action } from '@storybook/addon-actions';
import type { Meta, StoryFn } from '@storybook/react';
import type { ComponentType } from 'react';

import GenericModal, { GenericModalDoNotAskAgain } from './GenericModal';

export default {
	title: 'Components/GenericModal',
	component: GenericModal,
	subcomponents: { GenericModalDoNotAskAgain: GenericModalDoNotAskAgain as ComponentType<any> },
	args: {
		children: 'This is the content.',
	},
	parameters: {
		layout: 'fullscreen',
		controls: { hideNoControlsWarning: true },
		actions: { argTypesRegex: '^on.*' },
	},
} satisfies Meta<typeof GenericModal>;

export const Example: StoryFn<typeof GenericModal> = () => (
	<GenericModal
		title='Oh Myyy!'
		variant='warning'
		confirmText='Yay'
		cancelText='Nay'
		onConfirm={action('onConfirm')}
		onCancel={action('onCancel')}
		onClose={action('onClose')}
	>
		The quick brown fox jumps over the lazy dog.
	</GenericModal>
);

const Template: StoryFn<typeof GenericModal> = (args) => <GenericModal {...args} />;

export const Info = Template.bind({});
Info.args = { variant: 'info' };

export const Danger = Template.bind({});
Danger.args = { variant: 'danger' };

export const Warning = Template.bind({});
Warning.args = { variant: 'warning' };

export const Success = Template.bind({});
Success.args = { variant: 'success' };

export const WithDontAskAgain: StoryFn<typeof GenericModalDoNotAskAgain> = (args) => <GenericModalDoNotAskAgain {...args} />;
WithDontAskAgain.args = {
	dontAskAgain: {
		action: 'action',
		label: 'label',
	},
};
