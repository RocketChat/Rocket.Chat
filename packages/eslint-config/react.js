/** @type {import('eslint').ESLint.ConfigData} */
const config = {
	plugins: ['react', 'react-hooks', 'jsx-a11y'],
	extends: ['plugin:jsx-a11y/recommended'],
	rules: {
		'react-hooks/exhaustive-deps': 'error',
		'react-hooks/rules-of-hooks': 'error',
		'react/display-name': 'error',
		'react/jsx-curly-brace-presence': 'error',
		'react/jsx-fragments': ['error', 'syntax'],
		'react/jsx-key': ['error', { checkFragmentShorthand: true, checkKeyMustBeforeSpread: true, warnOnDuplicates: true }],
		'react/jsx-no-undef': 'error',
		'react/jsx-uses-react': 'error',
		'react/jsx-uses-vars': 'error',
		'react/no-multi-comp': 'error',
		'jsx-a11y/no-autofocus': [2, { ignoreNonDOM: true }],
	},
	settings: {
		react: {
			version: 'detect',
		},
	},
	overrides: [
		{
			files: ['**/*.stories.js', '**/*.stories.jsx', '**/*.stories.ts', '**/*.stories.tsx', '**/*.spec.tsx'],
			rules: {
				'react/display-name': 'off',
				'react/no-multi-comp': 'off',
			},
		},
	],
};

module.exports = config;
