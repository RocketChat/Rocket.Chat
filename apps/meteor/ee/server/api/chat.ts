import type { IMessage, ReadReceipt } from '@rocket.chat/core-typings';
import { License } from '@rocket.chat/license';
import { Meteor } from 'meteor/meteor';

import { API } from '../../../app/api/server/api';
import { getReadReceiptsFunction } from '../methods/getReadReceipts';

type GetMessageReadReceiptsProps = {
	messageId: IMessage['_id'];
};

declare module '@rocket.chat/rest-typings' {
	// eslint-disable-next-line @typescript-eslint/naming-convention
	interface Endpoints {
		'/v1/chat.getMessageReadReceipts': {
			GET: (params: GetMessageReadReceiptsProps) => {
				receipts: ReadReceipt[];
			};
		};
	}
}

API.v1.addRoute(
	'chat.getMessageReadReceipts',
	{ authRequired: true },
	{
		async get() {
			if (!License.hasModule('message-read-receipt')) {
				throw new Meteor.Error('error-action-not-allowed', 'This is an enterprise feature');
			}

			const { messageId } = this.queryParams;
			if (!messageId) {
				return API.v1.failure({
					error: "The required 'messageId' param is missing.",
				});
			}

			return API.v1.success({
				receipts: await getReadReceiptsFunction(messageId, this.userId),
			});
		},
	},
);
