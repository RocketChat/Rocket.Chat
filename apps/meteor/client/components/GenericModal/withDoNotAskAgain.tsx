import { Box, Label, CheckBox } from '@rocket.chat/fuselage';
import { useUniqueId } from '@rocket.chat/fuselage-hooks';
import { useUserPreference, useTranslation, useEndpoint } from '@rocket.chat/ui-contexts';
import type { ReactElement, ComponentType } from 'react';
import { useState } from 'react';

import type { DontAskAgainList } from '../../hooks/useDontAskAgain';

type DoNotAskAgainProps = {
	onConfirm: (...args: any) => Promise<void> | void;
	dontAskAgain: {
		action: string;
		label: string;
	};
};

export type RequiredModalProps = {
	onConfirm?: (...args: any) => Promise<void> | void;
	dontAskAgain?: ReactElement;
};

function withDoNotAskAgain<T extends RequiredModalProps>(Component: ComponentType<any>) {
	type WrappedComponentProps = DoNotAskAgainProps & Omit<T, keyof RequiredModalProps>;
	const WrappedComponent = function ({ onConfirm, dontAskAgain, ...props }: WrappedComponentProps) {
		const t = useTranslation();
		const dontAskAgainId = useUniqueId();
		const dontAskAgainList = useUserPreference<DontAskAgainList>('dontAskAgainList');
		const { action, label } = dontAskAgain;

		const saveFn = useEndpoint('POST', '/v1/users.setPreferences');
		const [state, setState] = useState<boolean>(false);

		const handleConfirm = async (): Promise<void> => {
			try {
				if (state) {
					await saveFn({ data: { dontAskAgainList: [...(dontAskAgainList || []), { action, label }] } });
				}
				await onConfirm();
			} catch (e) {
				console.error(e);
			}
		};

		const onChange = (): void => {
			setState(!state);
		};

		return (
			<Component
				{...props}
				dontAskAgain={
					<Box display='flex' flexDirection='row'>
						<CheckBox id={dontAskAgainId} checked={state} onChange={onChange} name='dont_ask_again' />
						<Label color='annotation' fontScale='p2' mis={8} htmlFor={dontAskAgainId}>
							{t('Dont_ask_me_again')}
						</Label>
					</Box>
				}
				onConfirm={handleConfirm}
			/>
		);
	};

	WrappedComponent.displayName = `withDoNotAskAgain(${Component.displayName ?? Component.name ?? 'Component'})`;

	return WrappedComponent;
}

export { withDoNotAskAgain };
