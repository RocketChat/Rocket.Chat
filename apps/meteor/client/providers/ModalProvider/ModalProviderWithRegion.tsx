import type { ReactNode } from 'react';
import { useState } from 'react';

import ModalProvider from './ModalProvider';
import ModalRegion from '../../views/modal/ModalRegion';

const ModalProviderWithRegion = ({ children }: { children?: ReactNode }) => {
	const [region] = useState(() => Symbol());

	return (
		<ModalProvider region={region}>
			<ModalRegion />
			{children}
		</ModalProvider>
	);
};

export default ModalProviderWithRegion;
