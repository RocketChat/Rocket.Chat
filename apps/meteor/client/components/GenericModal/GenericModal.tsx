import { Button, Modal } from '@rocket.chat/fuselage';
import { useEffectEvent, useUniqueId } from '@rocket.chat/fuselage-hooks';
import type { Keys as IconName } from '@rocket.chat/icons';
import type { ComponentProps, ReactElement, ReactNode, ComponentPropsWithoutRef } from 'react';
import { useEffect, useRef } from 'react';
import { useTranslation } from 'react-i18next';

import type { RequiredModalProps } from './withDoNotAskAgain';
import { withDoNotAskAgain } from './withDoNotAskAgain';

type VariantType = 'danger' | 'warning' | 'info' | 'success';

type GenericModalProps = RequiredModalProps & {
	variant?: VariantType;
	children?: ReactNode;
	cancelText?: ReactNode;
	confirmText?: ReactNode;
	title?: string | ReactElement;
	icon?: IconName | ReactElement | null;
	confirmDisabled?: boolean;
	tagline?: ReactNode;
	onCancel?: () => Promise<void> | void;
	onClose?: () => Promise<void> | void;
	onDismiss?: () => Promise<void> | void;
	annotation?: ReactNode;
} & Omit<ComponentPropsWithoutRef<typeof Modal>, 'title'>;

const iconMap: Record<string, IconName> = {
	danger: 'modal-warning',
	warning: 'modal-warning',
	info: 'info',
	success: 'check',
};

const getButtonProps = (variant: VariantType): ComponentProps<typeof Button> => {
	switch (variant) {
		case 'danger':
			return { danger: true };
		case 'warning':
			return { primary: true };
		default:
			return {};
	}
};

const renderIcon = (icon: GenericModalProps['icon'], variant: VariantType): ReactNode => {
	if (icon === null) {
		return null;
	}

	if (icon === undefined) {
		return <Modal.Icon color={variant} name={iconMap[variant]} />;
	}

	if (typeof icon === 'string') {
		return <Modal.Icon name={icon} />;
	}

	return icon;
};

const GenericModal = ({
	variant = 'info',
	children,
	cancelText,
	confirmText,
	title,
	icon,
	onCancel,
	onClose = onCancel,
	onDismiss = onClose,
	onConfirm,
	dontAskAgain,
	confirmDisabled,
	tagline,
	wrapperFunction,
	annotation,
	...props
}: GenericModalProps) => {
	const { t } = useTranslation();
	const genericModalId = useUniqueId();

	const dismissedRef = useRef(true);

	const handleConfirm = useEffectEvent(() => {
		dismissedRef.current = false;
		onConfirm?.();
	});

	const handleCancel = useEffectEvent(() => {
		dismissedRef.current = false;
		onCancel?.();
	});

	const handleCloseButtonClick = useEffectEvent(() => {
		dismissedRef.current = true;
		onClose?.();
	});

	useEffect(
		() => () => {
			if (!dismissedRef.current) return;
			onDismiss?.();
		},
		[onDismiss],
	);

	return (
		<Modal aria-labelledby={`${genericModalId}-title`} wrapperFunction={wrapperFunction} {...props}>
			<Modal.Header>
				{renderIcon(icon, variant)}
				<Modal.HeaderText>
					{tagline && <Modal.Tagline>{tagline}</Modal.Tagline>}
					<Modal.Title id={`${genericModalId}-title`}>{title ?? t('Are_you_sure')}</Modal.Title>
				</Modal.HeaderText>
				{onClose && <Modal.Close aria-label={t('Close')} onClick={handleCloseButtonClick} />}
			</Modal.Header>
			<Modal.Content fontScale='p2'>{children}</Modal.Content>
			<Modal.Footer justifyContent={dontAskAgain || annotation ? 'space-between' : 'end'}>
				{dontAskAgain}
				{annotation && !dontAskAgain && <Modal.FooterAnnotation>{annotation}</Modal.FooterAnnotation>}
				<Modal.FooterControllers>
					{onCancel && (
						<Button secondary onClick={handleCancel}>
							{cancelText ?? t('Cancel')}
						</Button>
					)}
					{wrapperFunction && (
						<Button {...getButtonProps(variant)} type='submit' disabled={confirmDisabled}>
							{confirmText ?? t('Ok')}
						</Button>
					)}
					{!wrapperFunction && onConfirm && (
						<Button {...getButtonProps(variant)} onClick={handleConfirm} disabled={confirmDisabled}>
							{confirmText ?? t('Ok')}
						</Button>
					)}
				</Modal.FooterControllers>
			</Modal.Footer>
		</Modal>
	);
};

export const GenericModalDoNotAskAgain = withDoNotAskAgain<GenericModalProps>(GenericModal);

export default GenericModal;
