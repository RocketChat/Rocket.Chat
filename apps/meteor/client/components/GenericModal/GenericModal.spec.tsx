import { useSetModal } from '@rocket.chat/ui-contexts';
import { act, screen, renderHook } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import type { ReactElement } from 'react';
import { Suspense } from 'react';

import GenericModal from './GenericModal';
import ModalProviderWithRegion from '../../providers/ModalProvider/ModalProviderWithRegion';

const renderModal = (modalElement: ReactElement) => {
	const {
		result: { current: setModal },
	} = renderHook(() => useSetModal(), {
		legacyRoot: true,
		wrapper: ({ children }) => (
			<Suspense fallback={null}>
				<ModalProviderWithRegion>{children}</ModalProviderWithRegion>
			</Suspense>
		),
	});

	act(() => {
		setModal(modalElement);
	});

	return { setModal };
};

describe('callbacks', () => {
	it('should call onClose callback when dismissed', async () => {
		const handleClose = jest.fn();

		renderModal(<GenericModal title='Modal' onClose={handleClose} />);

		expect(await screen.findByRole('heading', { name: 'Modal' })).toBeInTheDocument();

		await userEvent.keyboard('{Escape}');

		expect(screen.queryByRole('heading', { name: 'Modal' })).not.toBeInTheDocument();

		expect(handleClose).toHaveBeenCalled();
	});

	it('should NOT call onClose callback when confirmed', async () => {
		const handleConfirm = jest.fn();
		const handleClose = jest.fn();

		const { setModal } = renderModal(<GenericModal title='Modal' onConfirm={handleConfirm} onClose={handleClose} />);

		expect(await screen.findByRole('heading', { name: 'Modal' })).toBeInTheDocument();

		await userEvent.click(screen.getByRole('button', { name: 'Ok' }));

		expect(handleConfirm).toHaveBeenCalled();

		act(() => {
			setModal(null);
		});

		expect(screen.queryByRole('heading', { name: 'Modal' })).not.toBeInTheDocument();

		expect(handleClose).not.toHaveBeenCalled();
	});

	it('should NOT call onClose callback when cancelled', async () => {
		const handleCancel = jest.fn();
		const handleClose = jest.fn();

		const { setModal } = renderModal(<GenericModal title='Modal' onCancel={handleCancel} onClose={handleClose} />);

		expect(await screen.findByRole('heading', { name: 'Modal' })).toBeInTheDocument();

		await userEvent.click(screen.getByRole('button', { name: 'Cancel' }));

		expect(handleCancel).toHaveBeenCalled();

		act(() => {
			setModal(null);
		});

		expect(screen.queryByRole('heading', { name: 'Modal' })).not.toBeInTheDocument();

		expect(handleClose).not.toHaveBeenCalled();
	});
});
