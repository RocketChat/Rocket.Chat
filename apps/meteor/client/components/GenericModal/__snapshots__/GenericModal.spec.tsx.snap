// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Danger without crashing 1`] = `
<body>
  <div>
    <dialog
      aria-labelledby=":r0:-title"
      aria-modal="true"
      class="rcx-box rcx-box--full rcx-modal"
      open=""
    >
      <div
        class="rcx-box rcx-box--full rcx-modal__inner rcx-css-1e2ego0"
      >
        <header
          class="rcx-box rcx-box--full rcx-modal__header"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__header-inner"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-trljwa rcx-css-98qdbd"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-modal-warning rcx-icon rcx-css-zozvgz"
              >
                
              </i>
            </div>
            <div
              class="rcx-box rcx-box--full rcx-modal__header-text rcx-css-trljwa rcx-css-lma364"
            >
              <h2
                class="rcx-box rcx-box--full rcx-modal__title"
                id=":r0:-title"
              >
                Are_you_sure
              </h2>
            </div>
          </div>
        </header>
        <div
          class="rcx-box rcx-box--full rcx-modal__content rcx-css-1vw7itl"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-r1bpeb"
          >
            This is the content.
          </div>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-modal__footer rcx-css-17mu816"
        >
          <div
            class="rcx-button-group rcx-button-group--align-end"
            role="group"
          />
        </div>
      </div>
    </dialog>
  </div>
</body>
`;

exports[`renders Example without crashing 1`] = `
<body>
  <div>
    <dialog
      aria-labelledby=":r1:-title"
      aria-modal="true"
      class="rcx-box rcx-box--full rcx-modal"
      open=""
    >
      <div
        class="rcx-box rcx-box--full rcx-modal__inner rcx-css-1e2ego0"
      >
        <header
          class="rcx-box rcx-box--full rcx-modal__header"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__header-inner"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-trljwa rcx-css-1kjeq0v"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-modal-warning rcx-icon rcx-css-zozvgz"
              >
                
              </i>
            </div>
            <div
              class="rcx-box rcx-box--full rcx-modal__header-text rcx-css-trljwa rcx-css-lma364"
            >
              <h2
                class="rcx-box rcx-box--full rcx-modal__title"
                id=":r1:-title"
              >
                Oh Myyy!
              </h2>
            </div>
            <button
              aria-label="Close"
              class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button rcx-css-trljwa rcx-css-lma364"
              type="button"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-4pvxx3"
              >
                
              </i>
            </button>
          </div>
        </header>
        <div
          class="rcx-box rcx-box--full rcx-modal__content rcx-css-1vw7itl"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-r1bpeb"
          >
            The quick brown fox jumps over the lazy dog.
          </div>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-modal__footer rcx-css-17mu816"
        >
          <div
            class="rcx-button-group rcx-button-group--align-end"
            role="group"
          >
            <button
              class="rcx-box rcx-box--full rcx-button--secondary rcx-button rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Nay
              </span>
            </button>
            <button
              class="rcx-box rcx-box--full rcx-button--primary rcx-button rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Yay
              </span>
            </button>
          </div>
        </div>
      </div>
    </dialog>
  </div>
</body>
`;

exports[`renders Info without crashing 1`] = `
<body>
  <div>
    <dialog
      aria-labelledby=":r2:-title"
      aria-modal="true"
      class="rcx-box rcx-box--full rcx-modal"
      open=""
    >
      <div
        class="rcx-box rcx-box--full rcx-modal__inner rcx-css-1e2ego0"
      >
        <header
          class="rcx-box rcx-box--full rcx-modal__header"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__header-inner"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-trljwa rcx-css-3k19uf"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-info rcx-icon rcx-css-zozvgz"
              >
                
              </i>
            </div>
            <div
              class="rcx-box rcx-box--full rcx-modal__header-text rcx-css-trljwa rcx-css-lma364"
            >
              <h2
                class="rcx-box rcx-box--full rcx-modal__title"
                id=":r2:-title"
              >
                Are_you_sure
              </h2>
            </div>
          </div>
        </header>
        <div
          class="rcx-box rcx-box--full rcx-modal__content rcx-css-1vw7itl"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-r1bpeb"
          >
            This is the content.
          </div>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-modal__footer rcx-css-17mu816"
        >
          <div
            class="rcx-button-group rcx-button-group--align-end"
            role="group"
          />
        </div>
      </div>
    </dialog>
  </div>
</body>
`;

exports[`renders Success without crashing 1`] = `
<body>
  <div>
    <dialog
      aria-labelledby=":r3:-title"
      aria-modal="true"
      class="rcx-box rcx-box--full rcx-modal"
      open=""
    >
      <div
        class="rcx-box rcx-box--full rcx-modal__inner rcx-css-1e2ego0"
      >
        <header
          class="rcx-box rcx-box--full rcx-modal__header"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__header-inner"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-trljwa rcx-css-zlmdip"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-check rcx-icon rcx-css-zozvgz"
              >
                
              </i>
            </div>
            <div
              class="rcx-box rcx-box--full rcx-modal__header-text rcx-css-trljwa rcx-css-lma364"
            >
              <h2
                class="rcx-box rcx-box--full rcx-modal__title"
                id=":r3:-title"
              >
                Are_you_sure
              </h2>
            </div>
          </div>
        </header>
        <div
          class="rcx-box rcx-box--full rcx-modal__content rcx-css-1vw7itl"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-r1bpeb"
          >
            This is the content.
          </div>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-modal__footer rcx-css-17mu816"
        >
          <div
            class="rcx-button-group rcx-button-group--align-end"
            role="group"
          />
        </div>
      </div>
    </dialog>
  </div>
</body>
`;

exports[`renders Warning without crashing 1`] = `
<body>
  <div>
    <dialog
      aria-labelledby=":r4:-title"
      aria-modal="true"
      class="rcx-box rcx-box--full rcx-modal"
      open=""
    >
      <div
        class="rcx-box rcx-box--full rcx-modal__inner rcx-css-1e2ego0"
      >
        <header
          class="rcx-box rcx-box--full rcx-modal__header"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__header-inner"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-trljwa rcx-css-1kjeq0v"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-modal-warning rcx-icon rcx-css-zozvgz"
              >
                
              </i>
            </div>
            <div
              class="rcx-box rcx-box--full rcx-modal__header-text rcx-css-trljwa rcx-css-lma364"
            >
              <h2
                class="rcx-box rcx-box--full rcx-modal__title"
                id=":r4:-title"
              >
                Are_you_sure
              </h2>
            </div>
          </div>
        </header>
        <div
          class="rcx-box rcx-box--full rcx-modal__content rcx-css-1vw7itl"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-r1bpeb"
          >
            This is the content.
          </div>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-modal__footer rcx-css-17mu816"
        >
          <div
            class="rcx-button-group rcx-button-group--align-end"
            role="group"
          />
        </div>
      </div>
    </dialog>
  </div>
</body>
`;

exports[`renders WithDontAskAgain without crashing 1`] = `
<body>
  <div>
    <dialog
      aria-labelledby=":r6:-title"
      aria-modal="true"
      class="rcx-box rcx-box--full rcx-modal"
      open=""
    >
      <div
        class="rcx-box rcx-box--full rcx-modal__inner rcx-css-1e2ego0"
      >
        <header
          class="rcx-box rcx-box--full rcx-modal__header"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__header-inner"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-trljwa rcx-css-3k19uf"
            >
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-icon--name-info rcx-icon rcx-css-zozvgz"
              >
                
              </i>
            </div>
            <div
              class="rcx-box rcx-box--full rcx-modal__header-text rcx-css-trljwa rcx-css-lma364"
            >
              <h2
                class="rcx-box rcx-box--full rcx-modal__title"
                id=":r6:-title"
              >
                Are_you_sure
              </h2>
            </div>
          </div>
        </header>
        <div
          class="rcx-box rcx-box--full rcx-modal__content rcx-css-1vw7itl"
        >
          <div
            class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-r1bpeb"
          >
            This is the content.
          </div>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-modal__footer rcx-css-rzpqmp"
        >
          <div
            class="rcx-box rcx-box--full rcx-css-omrq7i"
          >
            <label
              class="rcx-box rcx-box--full rcx-check-box"
            >
              <input
                class="rcx-box rcx-box--full rcx-check-box__input"
                id=":r5:"
                name="dont_ask_again"
                type="checkbox"
              />
              <i
                aria-hidden="true"
                class="rcx-box rcx-box--full rcx-check-box__fake"
              />
            </label>
            <label
              class="rcx-box rcx-box--full rcx-label rcx-css-g9rc4a"
              for=":r5:"
            >
              Dont_ask_me_again
            </label>
          </div>
          <div
            class="rcx-button-group rcx-button-group--align-end"
            role="group"
          >
            <button
              class="rcx-box rcx-box--full rcx-button rcx-button-group__item"
              type="button"
            >
              <span
                class="rcx-button--content"
              >
                Ok
              </span>
            </button>
          </div>
        </div>
      </div>
    </dialog>
  </div>
</body>
`;
